File: patch.py
import os
import re

files_to_patch = {
    'app/api/ai/copilot/route.ts': [
        (r'where:\s*\{\s*userId\s*\}', r'where: { clientId: req.headers.get(\"x-client-id\") || \"legacy-client\" }'),
        (r'p\.analytics\.forEach\(a => \{', r'if (p.analytics) p.analytics.forEach((a: any) => {')
    ],
    'app/api/ai/image/route.ts': [
        (r'userId:\s*session\.user\.id,', r'clientId: req.headers.get(\"x-client-id\") || \"legacy-client\",')
    ],
    'app/api/auth/callback/facebook/route.ts': [
        (r'userId:\s*session\.user\.id,', r'clientId: \"legacy-client\",')
    ],
    'app/api/auth/callback/instagram/route.ts': [
        (r'userId:\s*session\.user\.id,', r'clientId: \"legacy-client\",')
    ],
    'app/api/auth/callback/linkedin/route.ts': [
        (r'userId:\s*session\.user\.id,', r'clientId: \"legacy-client\",')
    ],
    'app/api/auth/disconnect/route.ts': [
        (r'userId:\s*user\.id,', r'// removed userId')
    ],
    'app/api/auth/manual-connect/route.ts': [
        (r'userId:\s*session\.user\.id,', r'clientId: \"legacy-client\",')
    ],
    'app/api/auth/mock-connect/route.ts': [
        (r'userId,', r'clientId: \"legacy-client\",')
    ],
    'app/api/media/route.ts': [
        (r'where:\s*\{\s*userId\s*\}', r'where: { clientId: \"legacy-client\" }')
    ],
    'app/api/posts/comments/route.ts': [
        (r'post\.userId !== userId', r'post.clientId !== \"legacy-client\"')
    ],
    'app/api/posts/route.ts': [
        (r'userId:\s*session\.user\.id,', r'clientId: req.headers.get(\"x-client-id\") || \"legacy-client\",'),
        (r'post\.userId !== session\.user\.id', r'post.clientId !== (req.headers.get(\"x-client-id\") || \"legacy-client\")'),
        (r'where:\s*\{\s*userId:\s*session\.user\.id,\s*id:\s*mediaId\s*\}', r'where: { clientId: req.headers.get(\"x-client-id\") || \"legacy-client\", id: mediaId }')
    ],
    'app/api/upload/route.ts': [
        (r'userId:\s*session\.user\.id,', r'clientId: req.headers.get(\"x-client-id\") || \"legacy-client\",')
    ],
    'lib/queue.ts': [
        (r'userId:\s*post\.userId,', r'clientId: post.clientId,')
    ]
}

for filepath, patches in files_to_patch.items():
    if not os.path.exists(filepath):
        print(f'File not found: {filepath}')
        continue
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    for search, replace in patches:
        content = re.sub(search, replace, content)
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
    print(f'Patched {filepath}')

File: patch.js
const fs = require('fs');
const path = require('path');

const filesToPatch = {
    'app/api/ai/copilot/route.ts': [
        { search: /where:\s*\{\s*userId\s*\}/g, replace: 'where: { clientId: req.headers.get("x-client-id") || "legacy-client" }' },
        { search: /p\.analytics\.forEach\(a => \{/g, replace: 'if (p.analytics) p.analytics.forEach((a: any) => {' }
    ],
    'app/api/ai/image/route.ts': [
        { search: /userId:\s*session\.user\.id,/g, replace: 'clientId: req.headers.get("x-client-id") || "legacy-client",' }
    ],
    'app/api/auth/callback/facebook/route.ts': [
        { search: /userId:\s*session\.user\.id,/g, replace: 'clientId: "legacy-client",' }
    ],
    'app/api/auth/callback/instagram/route.ts': [
        { search: /userId:\s*session\.user\.id,/g, replace: 'clientId: "legacy-client",' }
    ],
    'app/api/auth/callback/linkedin/route.ts': [
        { search: /userId:\s*session\.user\.id,/g, replace: 'clientId: "legacy-client",' }
    ],
    'app/api/auth/disconnect/route.ts': [
        { search: /userId:\s*user\.id,/g, replace: '// removed userId' }
    ],
    'app/api/auth/manual-connect/route.ts': [
        { search: /userId:\s*session\.user\.id,/g, replace: 'clientId: "legacy-client",' }
    ],
    'app/api/auth/mock-connect/route.ts': [
        { search: /userId,/g, replace: 'clientId: "legacy-client",' }
    ],
    'app/api/media/route.ts': [
        { search: /where:\s*\{\s*userId\s*\}/g, replace: 'where: { clientId: "legacy-client" }' }
    ],
    'app/api/posts/comments/route.ts': [
        { search: /post\.userId !== userId/g, replace: 'post.clientId !== "legacy-client"' }
    ],
    'app/api/posts/route.ts': [
        { search: /userId:\s*session\.user\.id,/g, replace: 'clientId: req.headers.get("x-client-id") || "legacy-client",' },
        { search: /post\.userId !== session\.user\.id/g, replace: 'post.clientId !== (req.headers.get("x-client-id") || "legacy-client")' },
        { search: /where:\s*\{\s*userId:\s*session\.user\.id,\s*id:\s*mediaId\s*\}/g, replace: 'where: { clientId: req.headers.get("x-client-id") || "legacy-client", id: mediaId }' }
    ],
    'app/api/upload/route.ts': [
        { search: /userId:\s*session\.user\.id,/g, replace: 'clientId: req.headers.get("x-client-id") || "legacy-client",' }
    ],
    'lib/queue.ts': [
        { search: /userId:\s*post\.userId,/g, replace: 'clientId: post.clientId,' }
    ]
};

for (const [filepath, patches] of Object.entries(filesToPatch)) {
    const fullPath = path.join(process.cwd(), filepath);
    if (!fs.existsSync(fullPath)) {
        console.log('File not found:', filepath);
        continue;
    }
    let content = fs.readFileSync(fullPath, 'utf8');
    for (const { search, replace } of patches) {
        content = content.replace(search, replace);
    }
    fs.writeFileSync(fullPath, content, 'utf8');
    console.log('Patched', filepath);
}
